spring-cloud-data-module-registry/src/main/java/org/springframework/cloud/data/module/registry/RedisModuleRegistry.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.registry;
+
+import org.springframework.cloud.data.core.ModuleCoordinates;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisOperations;
+import org.springframework.data.redis.core.StringRedisTemplate;
+
+/**
+ * {@link ModuleRegistry} implementation backed by Redis.
+ *
+ * @author Patrick Peralta
+ */
+public class RedisModuleRegistry implements ModuleRegistry {
+
+	/**
+	 * Prefix for keys used for storing module coordinates.
+	 */
+	public static final String KEY_PREFIX = "spring.cloud.module";
+
+	/**
+	 * Group ID for default modules.
+	 */
+	private static final String GROUP_ID = "org.springframework.cloud.stream.module";
+
+	/**
+	 * Version number for default modules.
+	 */
+	private static final String VERSION = "1.0.0.BUILD-SNAPSHOT";
```

Comment:
If we keep this logic in the long run and s-c-data version is kept in synch with s-c-s-modules versions, would be nice to have this reflect the real version. Maybe we have found a good candidate for maven resource filtering in `application.yml` after all (pro-tip: use `@` as the delimiter)